docs/startup/preview/index.md

```diff
@@ -0,0 +1,529 @@
+2.  [Pageview Repartitioner](https://github.com/apache/samza-hello-samza/blob/c87ed565fbaebf2ac88376143c65e9f52f7a8801/src/main/java/samza/examples/cookbook/PageViewFilterApp.java) illustrates re-partitioning the incoming stream of PageViews.
+3.  [Pageview Sessionizer](https://github.com/apache/samza-hello-samza/blob/c87ed565fbaebf2ac88376143c65e9f52f7a8801/src/main/java/samza/examples/cookbook/PageViewSessionizerApp.java) groups the incoming stream of events into sessions based on user activity.
+4.  [Pageview by Region](https://github.com/apache/samza-hello-samza/blob/c87ed565fbaebf2ac88376143c65e9f52f7a8801/src/main/java/samza/examples/cookbook/TumblingPageViewCounterApp.java) counts the number of views per-region over tumbling time intervals.
+
+
+## Key concepts
+
+### StreamApplication
+When writing your stream processing application using the Samza high-level API, you should implement a [StreamApplication](/learn/documentation/{{site.version}}/api/javadocs/org/apache/samza/application/StreamApplication.html) and define your processing logic in the init method.
+{% highlight java %}
+public void init(StreamGraph graph, Config config) { … }
+{% endhighlight %}
+
+For example, here is a StreamApplication that validates and decorates page views with viewer's profile information.
+
+{% highlight java %}
+public class BadPageViewFilter implements StreamApplication {
+  @Override
+public void init(StreamGraph graph, Config config) {
+    MessageStream<PageView> pageViews = graph.getInputStream(“page-views”..);
+
+    pageViews.filter(this::isValidPageView)
+                      .map(this::addProfileInformation)
+                      .sendTo(graph.getOutputStream(“decorated-page-views”..))
+ }
+}
+{% endhighlight %}
+
+### MessageStream
+A MessageStream, as the name implies, represents a stream of messages. A StreamApplication is described as a series of transformations on MessageStreams. You can get a MessageStream in two ways:
+
+1. Using StreamGraph.getInputStream to get the MessageStream for a given input stream (e.g., a Kafka topic).
+
+2. By transforming an existing MessageStream using operations like map, filter, window, join etc.
+
+## Anatomy of a typical Samza StreamApplication
+There are 3 simple steps to write your stream processing applications using the Samza high-level API.
+
+### Step 1: Obtain the input streams:
+You can obtain the MessageStream for your input streamId (“page-views”) using StreamGraph.getInputStream.
+    {% highlight java %}
+    MessageStream<PageView> pageViewInput = graph.getInputStream(“page-views”, (k,v) -> v);
+    {% endhighlight %}
+
+The first parameter `page-views` is the `streamId` whose properties can be defined in configuration. For instance,
+
+{% highlight jproperties %}
+streams.PageViews.samza.system=kafka
```

Comment:
should be streams.page-views.... to be consistent with the code example above. 
I will fix it